tests/GeniusYield/Test/Providers/Mashup.hs

```diff
@@ -69,17 +72,24 @@ providersMashupTests configs =
           delayBySecond
           utxoAtRefWithDatum' <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxoAtTxOutRefWithDatum refWithDatumHash
           delayBySecond
-          utxosAtScriptCredential <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredential alwaysFailCredential
+          utxosAtScriptCredential <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredential alwaysFailCredential Nothing
           delayBySecond
-          utxosAtKeyCredential <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredential $ GYPaymentCredentialByKey "07fb2b78b3917d3f6bfa3a59de61f3c225cbf0d5564a1cbc6f96d6eb"  -- Credential of CI wallet.
+          utxosAtScriptCredentialWithAsset <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredential alwaysFailCredential (Just "6d24161a60592755dcbcc2c1330bbe968f913acc15ec40f0be3873ee.61757468")
           delayBySecond
-          utxosAtScriptCredentialWithDatums <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredentialWithDatums alwaysFailCredential
+          utxosAtKeyCredential <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredential ciWalletCredential Nothing
+          delayBySecond
+          utxosAtScriptCredentialWithDatums <- runGYTxQueryMonadNode (cfgNetworkId config) provider $ utxosAtPaymentCredentialWithDatums alwaysFailCredential Nothing
           delayBySecond
           utxosAtScriptAddressWithAsset <- gyQueryUtxosAtAddress provider alwaysFailAddress (Just "6d24161a60592755dcbcc2c1330bbe968f913acc15ec40f0be3873ee.61757468")  -- An asset I saw by random chance.
+          -- TODO: Write variant of above for with datums.
```

Comment:
I have deliberately left out test for this as it needs to be checked whether the corresponding test for _address_ (i.e. instead of payment credential) works now which earlier didn't due to some bug with BF.